terraform/modules/cluster/addons.tf

```diff
@@ -1,35 +1,17 @@
-data "aws_eks_addon_version" "latest" {
-  for_each = toset(["vpc-cni"])
-
-  addon_name         = each.value
-  kubernetes_version = var.cluster_version
-  most_recent        = true
-}
-
-resource "aws_eks_addon" "vpc_cni" {
-  cluster_name         = module.eks_blueprints.eks_cluster_id
-  addon_name           = "vpc-cni"
-  addon_version        = data.aws_eks_addon_version.latest["vpc-cni"].version
-  resolve_conflicts    = "OVERWRITE"
-  configuration_values = "{\"env\":{\"ENABLE_PREFIX_DELEGATION\":\"true\", \"ENABLE_POD_ENI\":\"true\"}}"
-
-  depends_on = [
-    null_resource.kubectl_set_env
-  ]
-}
-
 locals {
   ebs_csi_blocker = try(module.eks_blueprints_kubernetes_addons.aws_ebs_csi_driver.release_metadata.metadata.status, "")
 }
 
+resource "aws_iam_instance_profile" "karpenter" {
+  name_prefix = "karpenter-"
+  role        = module.eks.eks_managed_node_groups["managed-system"].iam_role_name
+  tags        = local.tags
+}
+
 module "eks_blueprints_kubernetes_addons" {
   source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.16.0//modules/kubernetes-addons"
 
-  depends_on = [
-    aws_eks_addon.vpc_cni
-  ]
-
-  eks_cluster_id = module.eks_blueprints.eks_cluster_id
+  eks_cluster_id = module.eks.cluster_name
```

Comment:
You will see a lot of `*cluster_id` references being replaced with `*cluster_name`. You can see more details about this change here https://github.com/hashicorp/terraform-provider-aws/issues/27560